stencil-library/src/components/payments-list/payments-list.tsx

```diff
@@ -2,69 +2,125 @@ import { Component, Host, h, Prop, State, Watch } from '@stencil/core';
 import { Api, IApiResponseCollection, Payment } from '../../api';
 import { formatCurrency, formatDate, formatTime } from '../../utils/utils';
 
+/**
+   *
+   * @exportedPart table-head: Head of the table
+   * @exportedPart table-body: Body of the table
+   * @exportedPart loading-state: Row for loading state
+   * @exportedPart error-state: Row for Error state
+   * @exportedPart empty-state: Row for Emtpy state
+   *
+   */
 @Component({
   tag: 'justifi-payments-list',
-  styleUrl: 'payments-list.css',
+  styleUrl: 'payments-list.scss',
   shadow: true,
 })
+
 export class PaymentsList {
   @Prop() accountId: string;
-  @Prop() auth: { token?: string } = {};
+  @Prop() authToken: string;
   @State() payments: Payment[] = [];
+  @State() loading: boolean = true;
+  @State() errorMessage: string;
 
   @Watch('accountId')
-  @Watch('auth')
-  requestPropsChanged() {
+  @Watch('authToken')
+  updateOnPropChange() {
     this.fetchData();
   }
 
-  fetchData(): void {
-    const accountId = this.accountId;
-    const endpoint = `account/${accountId}/payments`;
-    Api(this.auth.token)
-      .get(endpoint)
-      .then((response: IApiResponseCollection<Payment[]>) => {
-        const data = response.data.map(dataItem => new Payment(dataItem));
-        this.payments = data;
-      });
+  connectedCallback() {
+    this.fetchData();
   }
 
+  async fetchData(): Promise<void> {
+    if (!this.accountId || !this.authToken) {
+      this.errorMessage = "Can not fetch any data without an AccountID and an AuthToken"
+      this.loading = false;
+      return;
+    }
+    this.loading = true;
+    const endpoint = `account/${this.accountId}/payments`;
+
+    const response: IApiResponseCollection<Payment[]> = await Api(this.authToken).get(endpoint);
+    if (!response.error) {
+      const data = response?.data?.map(dataItem => new Payment(dataItem));
+      this.payments = data;
+    } else {
+      this.errorMessage = typeof response.error === 'string' ? response.error : response.error.message;
+    }
+
+    this.loading = false;
+  }
+
+  showEmptyState() {
+    return this.payments ? this.payments.length < 1 : true;
+  }
+
+  emptyState = (
+    <tr part="empty-state">
```

Comment:
Have you tried using these parts to apply styling? I believe you cannot do ```justifi-payment-form::part(empty-state) td {...}``` for example, meaning you cannot select the child of a part, but only the part directly.